2.0.0 build failure - mismatched types

Hi,

Trying to update our package (NetBSD) to 2.0.0 fails with:

```
   Compiling erdtree v2.0.0 (/usr/pkgsrc/wip/erdtree/work/erdtree-2.0.0)
warning: unused import: `ptr`
 --> src/fs/xattr.rs:2:48
  |
2 | use std::{os::unix::ffi::OsStrExt, path::Path, ptr};
  |                                                ^^^
  |
  = note: `#[warn(unused_imports)]` on by default

warning: unused import: `listxattr`
  --> src/fs/xattr.rs:25:24
   |
25 |     use libc::{c_char, listxattr};
   |                        ^^^^^^^^^

error[E0308]: mismatched types
  --> src/fs/xattr.rs:24:38
   |
24 | unsafe fn has_xattrs(path: &Path) -> bool {
   |           ----------                 ^^^^ expected `bool`, found `()`
   |           |
   |           implicitly returns `()` as its body has no tail or `return` expression

For more information about this error, try `rustc --explain E0308`.
warning: `erdtree` (bin "erd") generated 2 warnings
error: could not compile `erdtree` due to previous error; 2 warnings emitted
*** Error code 101

Stop.
```

OS: NetBSD-current (development branch)
Rust: 1.69.0

--- a/src/fs/xattr.rs
+++ b/src/fs/xattr.rs
@@ -30,7 +30,7 @@ unsafe fn has_xattrs(path: &Path) -> bool {
         slice_ptr.cast::<c_char>()
     };
 
-    #[cfg(target_os = "linux")]
+    #[cfg(not(target_os = "macos"))]
     return 0 < listxattr(path_ptr, ptr::null_mut::<c_char>(), 0);
 
     #[cfg(target_os = "macos")]

